File: backend/bolts/_http_models.py
import datetime
from typing import Literal

import pydantic


class PutTextRequest(pydantic.BaseModel):
    text: str


class GetTextResponse(pydantic.BaseModel):
    text: str
    lastModifiedAt: datetime.datetime


ButtonID = Literal["red", "green", "blue", "yellow"]


class PostButtonRequest(pydantic.BaseModel):
    pass


class GetButtonResponse(pydantic.BaseModel):
    timesClicked: int


class PostPostRequest(pydantic.BaseModel):
    title: str
    body: str


class GetPostResponse(pydantic.BaseModel):
    id: str
    title: str
    body: str


class PostTitleOnly(pydantic.BaseModel):
    title: str


class GetAllPostsResponse(pydantic.BaseModel):
    data: list[GetPostResponse]


class GetAllPostsResponseTitlesOnly(pydantic.BaseModel):
    data: list[PostTitleOnly]

File: frontend/src/apiClient/httpModels.ts
// JSON shapes for HTTP requests and responses.

import { z } from "zod";

export type ButtonID = "red" | "green" | "blue" | "yellow";

export const postButtonRequestSchema = z.object({});
export type PostButtonRequest = z.infer<typeof postButtonRequestSchema>;

export const getButtonResponseSchema = z.object({
  timesClicked: z.number(),
});
export type GetButtonResponse = z.infer<typeof getButtonResponseSchema>;

export const putTextRequestSchema = z.object({
  text: z.string(),
});
export type PutTextRequest = z.infer<typeof putTextRequestSchema>;

export const getTextResponseSchema = z.object({
  text: z.string(),
  lastModifiedAt: z.string(),
});
export type GetTextResponse = z.infer<typeof getTextResponseSchema>;
